tests/testthat/test-coverage.R

```diff
@@ -90,3 +90,27 @@ test_that("emdn_get_coverage() works -- stack", {
     )
   )
 })
+
+test_that("`nil_values_as_na = TRUE` converts NaN nil values to NA", {
+  # https://github.com/EMODnet/emodnet.wcs/issues/121
+  vcr::local_cassette("nil-values")
+
+  hab_wcs <- emdn_init_wcs_client(service = "seabed_habitats")
+  coverage_id <- "emodnet_open_maplibrary__GB000050_EFH_Whiting_SpawningGrounds"
+  test_bbox <- c(xmin = 0, ymin = 55, xmax = 2, ymax = 57)
+
+  # no conversion
+  summary <- emdn_get_coverage_summaries(hab_wcs, coverage_id)[[1]]
+  expect_equal(emdn_get_band_nil_values(summary), c(GRAY_INDEX = NaN))
+
+  # conversion
+  expect_snapshot(
+    r <- emdn_get_coverage(
+      wcs = hab_wcs,
+      coverage_id = coverage_id,
+      bbox = test_bbox,
+      nil_values_as_na = TRUE
+    )
+  )
```

Comment:
Not sure this snapshot is really useful for testing this feature

Reply:
I wanted to catch the verbosity.